dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -2461,22 +2458,15 @@ int ha_mcs_impl_rnd_end(TABLE* table, bool is_pushdown_hand)
                 thd->lex->sql_command == SQLCOM_LOAD))
         return 0;
 
-
-    // MCOL-2178 isUnion member only assigned, never used
-    //    if (is_pushdown_hand)
-    //    {
-    //        MIGR::infinidb_vtable.isUnion = false;
-    //    }
-
     if (get_fe_conn_info_ptr() != NULL)
         ci = reinterpret_cast<cal_connection_info*>(get_fe_conn_info_ptr());
     if ( (thd->lex)->sql_command == SQLCOM_ALTER_TABLE )
         return rc;
 
-    if (((thd->lex)->sql_command == SQLCOM_UPDATE)  ||
-            ((thd->lex)->sql_command == SQLCOM_DELETE) ||
-            ((thd->lex)->sql_command == SQLCOM_DELETE_MULTI) ||
-            ((thd->lex)->sql_command == SQLCOM_UPDATE_MULTI))
+    if ((thd->lex->sql_command == SQLCOM_UPDATE) ||
```

Comment:
Is it possible to replace this repeating conditions with the bool func you created?

Reply:
Not possible in this case since the following `&&` goes together:
`((thd->lex->sql_command == SQLCOM_UPDATE_MULTI) && !isForeignTableUpdate(thd))`